File: app/src/main/java/com/djacoronel/basiclauncher/iconutils/IconPickerAdapter.java
package com.djacoronel.basiclauncher.iconutils;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import com.djacoronel.basiclauncher.R;

import java.util.List;

class IconPickerAdapter extends RecyclerView.Adapter<IconPickerAdapter.ViewHolder> {
    private List<String> icons;
    private String iconPack;
    private Context mContext;

    IconPickerAdapter(List<String> icons, String iconPack, Context context) {
        this.icons = icons;
        this.iconPack = iconPack;
        mContext = context;
    }

    class ViewHolder extends RecyclerView.ViewHolder {
        ImageView icon;

        ViewHolder(View itemView) {
            super(itemView);
            icon = (ImageView) itemView.findViewById(R.id.item_app_icon);
            itemView.setOnClickListener(
                    new View.OnClickListener() {
                        @Override
                        public void onClick(View v) {
                            ((IconsActivity) mContext).pickIcon(icons.get(getAdapterPosition()));
                        }
                    }
            );
        }
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.app_detail, parent, false);
        return new ViewHolder(v);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        IconPackManager icManager = new IconPackManager(mContext);
        Drawable icon = icManager.loadDrawable(icons.get(position), iconPack);
        if (icon == null) icons.remove(position);
        else holder.icon.setImageDrawable(icon);
    }

    @Override
    public int getItemCount() {
        return icons.size();
    }
}

File: app/src/main/java/com/djacoronel/sleeklauncher/iconutils/IconPickerAdapter.kt
package com.djacoronel.sleeklauncher.iconutils

import android.content.Context
import android.graphics.drawable.Drawable
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.djacoronel.sleeklauncher.R
import kotlinx.android.synthetic.main.app_detail.view.*

class IconPickerAdapter(
        private val icons: List<String>,
        private val iconPack: String,
        private val mContext: Context
) : RecyclerView.Adapter<IconPickerAdapter.ViewHolder>() {

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        fun bind(icon: Drawable) {
            itemView.item_app_icon.setImageDrawable(icon)
            itemView.setOnClickListener { (mContext as IconsActivity).setIconAsResult(icons[adapterPosition]) }
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val v = LayoutInflater.from(parent.context)
                .inflate(R.layout.app_detail, parent, false)
        return ViewHolder(v)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val icManager = IconPackManager(mContext)
        val icon = icManager.loadDrawable(icons[position], iconPack)
        icon?.let {
            holder.bind(it)
        }
    }

    override fun getItemCount(): Int = icons.size
}
